mono/mini/interp/transform.c

```diff
@@ -4503,6 +4503,17 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 				td->last_ins->data [0] = get_data_item_index (td, mono_interp_get_imethod (domain, m, error));
 				goto_if_nok (error, exit);
 
+				PUSH_TYPE (td, stack_type [mint_type (m_class_get_byval_arg (klass))], klass);
+			} else if (klass == mono_defaults.int_class && csignature->param_count == 1)  {
+				td->sp--;
+#if SIZEOF_VOID_P == 8
+				if (td->sp [0].type == STACK_TYPE_I4)
+					interp_add_ins (td, MINT_CONV_I8_I4);
+#else
+				if (td->sp [0].type == STACK_TYPE_I8)
+					interp_add_ins (td, MINT_CONV_OVF_I4_I8);
+#endif
```

Comment:
There are some other ctors in IntPtr.

Reply:
This is meant to happen ctors from `int`, `long` and `void*` why seem the most common. Any other that you think would warrant optimization ?

Reply:
I mean shouldn't we emit some other default opcode in those cases ?